### PR TITLE
Feature(jdbc-catalog)：Add code skeleton for oceanbase-ce jdbc catalog

### PR DESCRIPTION
**Pull request by yuanoOo**
### What changes were proposed in this pull request?

- add catalog-jdbc-oceanbase module
- add code skeleton for oceanbase-ce jdbc catalog

### Why are the changes needed?
Fix: #4916 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?
build pass. tests will be added in implementation PRs

### CODE REVIEW

**Comment by yuanoOo** [comment]
@mchades PTAL.

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/OceanBaseCatalogCapability.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * under the License.
+ */
+package org.apache.gravitino.catalog.oceanbase;
+
+import org.apache.gravitino.connector.capability.Capability;
+import org.apache.gravitino.connector.capability.CapabilityResult;
+
+public class OceanBaseCatalogCapability implements Capability {
+
+  public static final String OCEANBASE_NAME_PATTERN = "^[\\w\\p{L}-$/=]{1,64}$";
```
Comment (mchades) [inline catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/OceanBaseCatalogCapability.java:26]: Could you plz add a comment for the regular expression？

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseColumnDefaultValueConverter.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import org.apache.gravitino.catalog.jdbc.converter.JdbcColumnDefaultValueConverter;
+import org.apache.gravitino.catalog.jdbc.converter.JdbcTypeConverter;
+import org.apache.gravitino.rel.expressions.Expression;
+import org.apache.gravitino.rel.expressions.UnparsedExpression;
+import org.apache.gravitino.rel.expressions.literals.Literals;
+
+public class OceanBaseColumnDefaultValueConverter extends JdbcColumnDefaultValueConverter {
+
+  @Override
+  public Expression toGravitino(
```
Comment (mchades) [inline catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseColumnDefaultValueConverter.java:34]: Please remove the implementation as it is not needed for this PR.

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseTypeConverter.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+/** Type converter for OceanBase. */
+public class OceanBaseTypeConverter extends JdbcTypeConverter {
+
+  // More data types will be added.
+  static final String TINYINT = "tinyint";
+
+  @Override
+  public Type toGravitino(JdbcTypeBean typeBean) {
+    switch (typeBean.getTypeName().toLowerCase()) {
```
Comment (mchades) [inline catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseTypeConverter.java:33]: Same here,  the implementation is unnecessary in this PR

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/operation/OceanBaseDatabaseOperations.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    throw new UnsupportedOperationException("Not implemented yet.");
+  }
+
+  @Override
+  public String generateDropDatabaseSql(String databaseName, boolean cascade) {
+    throw new UnsupportedOperationException("Not implemented yet.");
+  }
+
+  @Override
+  public JdbcSchema load(String databaseName) throws NoSuchSchemaException {
```
Comment (mchades) [inline catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/operation/OceanBaseDatabaseOperations.java:60]: Same here

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/operation/OceanBaseTableOperations.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+import org.apache.gravitino.rel.expressions.distributions.Distribution;
+import org.apache.gravitino.rel.expressions.transforms.Transform;
+import org.apache.gravitino.rel.indexes.Index;
+
+/** Table operations for OceanBase. */
+public class OceanBaseTableOperations extends JdbcTableOperations {
+
+  @Override
+  public List<String> listTables(String databaseName) throws NoSuchSchemaException {
+    final List<String> names = Lists.newArrayList();
```
Comment (mchades) [inline catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/operation/OceanBaseTableOperations.java:44]: Same here

**Review by mchades — COMMENTED** [review]
Plz remove implementation in this PR since it's for code skeleton only

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/OceanBaseCatalogCapability.java (hunk 1 above)
Comment (yuanoOo) [reply catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/OceanBaseCatalogCapability.java:41]: done.

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseTypeConverter.java (hunk 3 above)
Comment (yuanoOo) [reply catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseTypeConverter.java]: done.

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/operation/OceanBaseDatabaseOperations.java (hunk 4 above)
Comment (yuanoOo) [reply catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/operation/OceanBaseDatabaseOperations.java:57]: done.

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/operation/OceanBaseTableOperations.java (hunk 5 above)
Comment (yuanoOo) [reply catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/operation/OceanBaseTableOperations.java]: done.

File: catalogs/catalog-jdbc-oceanbase/src/test/resources/log4j2.properties [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+name = ConsoleLogConfig
+
+# Console appender configuration
+appender.console.type = Console
+appender.console.name = consoleLogger
+appender.console.layout.type = PatternLayout
+appender.console.layout.pattern = %d{yyyy-MM-dd HH:mm:ss} %-5p [%t] %c{1}:%L - %m%n
+
+# Log files location
+property.logPath = ${sys:gravitino.log.path:-build/catalog-jdbc-mysql-integration-test.log}
```
Comment (mchades) [inline catalogs/catalog-jdbc-oceanbase/src/test/resources/log4j2.properties:33]: should be `build/catalog-jdbc-oceanbase-integration-test.log`

Comment (yuanoOo) [reply catalogs/catalog-jdbc-oceanbase/src/test/resources/log4j2.properties]: Thanks, fixed.

**Review by mchades — APPROVED** [review]
LGTM. Thanks for your contributions!

I'm going to merge this PR, and I think the next PR should be `support schema operations for oceanbase catalog`

**Comment by yuanoOo** [comment]
OK, I will create a subtask issue to track the upcoming schema operation PR.

**Pull request merged**